### PR TITLE
[github_search] rename and consolidate

### LINKED ISSUE #146: github_list user search - add stats to the links
**Is your feature request related to a problem? Please describe.**

github_list user search doesn't give enough insight into the user's profile

**Describe the solution you'd like**
add numbers alongside the links 
e.g.
[repositories](https://github.com/rabbah?tab=repositories) **71**
[projects](https://github.com/rabbah?tab=projects) **0**
[stars](https://github.com/rabbah?tab=stars)  **67**
[following](https://github.com/rabbah?tab=following) **0** 
[followers](https://github.com/rabbah?tab=followers) **59**

**Additional context**
Screenshot from 2020-04-18 23-50-26

### LINKED ISSUE #177: github list should instead be github search

### PR DESCRIPTION
**Pull request by bhageena**
fixes #177 
fixes #146

### CODE REVIEW

**Comment by satyarohith** [comment]
I will review in the morning. Please do not merge yet.

**Review by satyarohith — COMMENTED** [review]
`github_view_pr` shows the recent pull requests of a repository and `github_find_pr` shows PRs updated on a specific date. Since they provide different features it might be better to keep them.

File: github/packages/github/github_search.js [hunk 1]
```diff
@@ -22,85 +16,114 @@ async function getRequest(url, secrets) {
...
       entity = 'code';
       displayEntity = 'Code Files';
-      keywords += '+in:file';
-      if (!repository) return fail('*please specify a repository, using -r flag e.g.*\n`/nc github_list c github -r nimbella/command-sets`');
+      query += '+in:file';
+      if (!keywords) return fail('*please specify a keyword*')
+      if (!repositories) return fail('*please specify a repository, using -r flag e.g.*\n`/nc github_search c github -r nimbella/command-sets`');
       break;
     case 'i':
     case 'issue':
```
Comment (rabbah) [inline github/packages/github/github_search.js:73]: add 'issues'
```suggestion
    case 'issue':
    case 'issues':
```

File: github/packages/github/github_search.js [hunk 2]
```diff
@@ -22,85 +16,114 @@ async function getRequest(url, secrets) {
...
     case 'u':
     case 'user':
       displayEntity = 'Users';
       entity = 'users';
+      repositories = undefined;
+      adjustedPageSize = 10;
+      if (!keywords) return fail('*please specify a keyword*')
       break;
     case 't':
     case 'topic':
```
Comment (rabbah) [inline github/packages/github/github_search.js:98]: add `topics`
```suggestion
    case 'topic':
    case 'topics':
```

**Comment by rabbah** [comment]
Can I simply list issues/prs with search (understanding I need pagination in general)?

**Comment by bhageena** [comment]
> Can I simply list issues/prs with search (understanding I need pagination in general)?

yes, pagination is optional

**Comment by bhageena** [comment]
> `github_view_pr` shows the recent pull requests of a repository and `github_find_pr` shows PRs updated on a specific date. Since they provide different features it might be better to keep them.

This implementation of `github_search` also incorporates functionalities offered through  `github_view_pr` and `github_find_pr`. I added back the `github_view_pr` just in case someone is using it, but it's not part of the docs on purpose. We'll look at the usage metrics and remove it, if safe.

**Review by niks3089 — APPROVED** [review]

**Pull request merged**